### PR TITLE
DO NOT MERGE oauth2-provider-web

### PR DESCRIPTION
**Pull request by csierra**

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+			for (CheckPattern pattern : _patterns) {
+				Predicate<String> urlPatternPredicate =
+					pattern.getUrlPatternPredicate();
+
+				if (urlPatternPredicate.test(path)) {
+					Predicate<String> methodPatternPredicate =
+						pattern.getMethodPatternPredicate();
+
+					if (methodPatternPredicate.test(request.getMethod())) {
+						if (_scopeChecker.checkAnyScope(pattern.getScopes())) {
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: Please change for `checkAllScopes`

Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: Please allow empty scopes
1, when the URL/method matches and `scopes` is empty then just return ... means `@RequiresNoScope`
2, don't forget to filter out empty scope from the ScopeFinder

Thanks!

Comment (topolik) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: `checkAnyScope` can be done using multiple same lines/patterns

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+			Request request = containerRequestContext.getRequest();
+
+			String path = _uriInfo.getPath();
+
+			for (CheckPattern pattern : _patterns) {
+				Predicate<String> urlPatternPredicate =
+					pattern.getUrlPatternPredicate();
+
+				if (urlPatternPredicate.test(path)) {
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: Consider using `StringUtil.wildcardMatches`, Shuyang said it's simpler but faster.

**Comment by topolik** [comment]
Please resend, thanks!

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java (hunk 2 above)
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: it wouldn't be the same... would it?

Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: we would miss the `|` operator

Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: if people really uses this and we want to make it really fast I guess we should implement a trie with a longest prefix match, or stuff like that

Comment (topolik) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/ConfigurableCheckerFeature.java]: Please don't kill me! I only said "consider" :smile:

**Pull request closed without merging**